easybuild/framework/easyblock.py

```diff
@@ -3350,6 +3353,365 @@ def _sanity_check_step_multi_deps(self, *args, **kwargs):
+            trace_msg(short_msg)
+            msg = "You may consider rerunning with --cuda-sanity-check-accept-ptx-as-devcode to accept "
+            msg += "binaries that don't contain the device code for your requested CUDA Compute Capabilities, "
+            msg += "but that do have PTX code that can be compiled for your requested CUDA Compute "
+            msg += "Capabilities. Note that this may increase startup delay due to JIT compilation "
+            msg += "and may also lead to suboptimal runtime performance, as the PTX code may not exploit "
+            msg += "all features specific to your hardware architecture."
+            self.log.info(msg)
+        if len(files_additional_devcode) > 0 and strict_cc_check:
+            short_msg = "You may consider running with --disable-cuda-sanity-check-strict to accept binaries "
+            short_msg += "containing device code for more architectures than requested."
+            trace_msg(short_msg)
+            msg = "You may consider running with --disable-cuda-sanity-check-strict. This means you'll "
+            msg += "accept that some binaries may have CUDA Device Code for more architectures than you "
+            msg += "requested, i.e. the binary is 'fatter' than you need. Bigger binaries may generally "
+            msg += "cause some startup delay, and code path selection could introduce a small overhead, "
+            msg += "though this is generally negligible."
+            self.log.info(msg)
+        if len(files_missing_ptx) > 0 and not accept_missing_ptx:
+            short_msg = "You may consider running with --cuda-sanity-check-accept-missing-ptx to accept binaries "
+            short_msg += "missing PTX code for the highest configured CUDA Compute Capability."
+            trace_msg(short_msg)
+            msg = "You may consider running with --cuda-sanity-check-accept-missing-ptx to accept binaries "
+            msg += "that don't contain PTX code for the highest CUDA Compute Capability you requested. This "
+            msg += "breaks forwards compatibility for newer CUDA Compute Capabilities (i.e. your compiled "
+            msg += "binaries will not run on cards with higher CUDA Compute Capabilities than what "
+            msg += "you requested in --cuda-compute-capabilities), but that may be acceptable to you."
+            self.log.info(msg)
+        if (
+            build_option('debug') and
+            (len(files_missing_devcode) > 0 or len(files_additional_devcode) > 0 or len(files_missing_ptx) > 0)
+        ):
+            trace_and_log("See build log for detailed lists of files not passing the CUDA Sanity Check")
+        elif (
+            len(files_missing_devcode) > 0 or len(files_additional_devcode) > 0 or len(files_missing_ptx) > 0
+        ):
+            msg = "To get a detailed list of files not passing the CUDA Sanity Check in the build log, "
+            msg += "rerun with --debug."
+            trace_and_log(msg)
```

Comment:
This is confusing, it seems the list is already in the log from 3711 below so only the first condition is needed (minus the `build_option` check)?
```suggestion
        if (
            len(files_missing_devcode) > 0 or len(files_additional_devcode) > 0 or len(files_missing_ptx) > 0
        ):
            trace_and_log("See build log for detailed lists of files not passing the CUDA Sanity Check")
```